EIPS/eip-8037.md

```diff
@@ -31,62 +31,133 @@ The relationship we are seeing in this example is not linear as expected. This i
+| :-----------------------------: | :---------------: | :-----------: | :----------: | :----------------------------------------: |
+|       `CREATE_FIXED_COST`       |      32,000       |    212,800    |     6.7x     | `CREATE`, `CREATE2`, contract creation txs |
+|       `CODE_DEPOSIT_GAS`        |        200        |     1,900     |     9.5x     | `CREATE`, `CREATE2`, contract creation txs |
+|     `ACCOUNT_CREATION_COST`     |      25,000       |    212,800    |     8.5x     |  New account creation (EOA and contracts)  |
+| `GAS_SELF_DESTRUCT_NEW_ACCOUNT` |      25,000       |    212,800    |     8.5x     |              `SELF_DESTRUCT`               |
+|        `GAS_STORAGE_SET`        |      20,000       |    60,800     |      3x      |                  `SSTORE`                  |
+|    `PER_EMPTY_ACCOUNT_COST`     |      25,000       |    212,800    |     8.5x     |               EOA delegation               |
+|      `PER_AUTH_BASE_COST`       |      12,500       |    43,700     |     3.5x     |               EOA delegation               |
+|           `HASH_GAS`            |         -         |  6 per word   |      -       |   Runtime code hashing for deduplication   |
 
 ### Multidimensional metering for code deposit gas
 
 Besides the parameter changes, this proposal introduces an independent metering for the code deposit costs. The specification is derived from [EIP-8011](./eip-8011.md). However, it only requires two dimensions, namely, `gas` and `code_deposit_gas`.
 
 ### Contract deployment cost calculation
 
-This proposal further changes how contract deployment cost are computed. When a contract creation returns a runtime bytecode `R` (length `L`), first check whether the code already exists in the state trie. Then:
-
-- If `CodeExists(keccak256(R), live_state) == false`, charge `code_deposit_cost=GAS_CODE_DEPOSIT*L` and store `R` under its code hash.
-- If `CodeExists(...) == true`, do not charge code-deposit storage gas (`code_deposit_gas=0`); simply link the new account's `codeHash` to the existing code object.
-
-In addition, contract creation is also charged a `storage_access_cost = GAS_WARM_ACCESS (if warm) | GAS_COLD_SLOAD (if cold)` and a `hash_cost = GAS_KECCAK256_WORD * code_data_words`, where `code_data_words = ceil(L / 32)`. This accounts for the added execution cost of accessing and verifying for duplicated code.
+This proposal fundamentally changes how contract deployment costs are computed to properly account for success and failure paths, ensuring that post-execution costs are only charged when deployment succeeds.
+
+#### Success vs Failure Gas Accounting
+
+When a contract creation transaction or opcode (`CREATE`/`CREATE2`) is executed, gas is charged differently based on whether the deployment succeeds or fails. Given runtime bytecode `R` (length `L`) returned by initcode and `H = keccak256(R)`:
```

Comment:
runtime and `R` are not very good names imo. I would drop "runtime", as the rest of the sentence specifies that this is not the initcode. And then rename `R` to `B` so that it matches the name and isn't confusing to read.